File: ADVANCED BUSINESS DEVELOPMENT WITH .NET/src/Domain/Entities/Cliente.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace UserApi.Domain
    {
        public class Cliente
        {
            [BsonId]
            public ObjectId Id { get; set; }
            public string? IdCliente { get; set; }
            public string Nome { get; set; } = null!;
            public string Sobrenome { get; set; } = null!;
            public string Email { get; set; } = null!;
            public string Telefone { get; set; } = null!;
            public string Endereco { get; set; } = null!;
            public DateTime DataNasc { get; set; }
            public string Senha { get; set; } = null!;

        }
    }

File: JAVA ADVANCED/sprint-2/challenge/src/main/java/br/com/fiap/challenge/domains/Cliente.java
package br.com.fiap.challenge.domains;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.UniqueElements;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "cliente")
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id_cliente", nullable = false)
    private String idCliente;

    @Column(name = "nome", length = 100, nullable = false)
    private String nome;

    @Column(name = "sobrenome", length = 100, nullable = false)
    private String sobrenome;

    @Column(name = "email", length = 100, nullable = false)
    private String email;

    @Column(name = "telefone", length = 15, nullable = false)
    private String telefone;

    @Column(name = "data_nasc", nullable = false)
    private LocalDate dataNasc;

    @Column(name = "endereco", length = 255, nullable = false)
    private String endereco;

}
